pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,465 @@
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=2)
+
+            drawer.box_gate(layer=0, wires=0, text="Y")
+            drawer.box_gate(layer=1, wires=(0,1), text="CRy(0.1)", rotate_text=True)
+
+        .. figure:: ../../_static/drawer/box_gates.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        wires = _to_tuple(wires)
+
+        box_min = min(wires)
+        box_max = max(wires)
+        box_len = box_max - box_min
+        box_center = (box_max + box_min) / 2.0
+
+        if rotate_text:
+            rotation = "vertical"
+        else:
+            rotation = "horizontal"
+
+        box = plt.Rectangle(
+            (layer - self._box_dx - extra_width / 2, box_min - self._box_dx),
+            2 * self._box_dx + extra_width,
+            (box_len + 2 * self._box_dx),
+            zorder=2 + zorder_base,
+            facecolor=color,
+        )
+        self.ax.add_patch(box)
+
+        self.ax.text(
+            layer,
+            box_center,
+            text,
+            zorder=3 + zorder_base,
+            rotation=rotation,
+            ha="center",
+            va="center",
+            fontsize="x-large",
+        )
+
+    def ctrl(self, layer, wire_ctrl, wire_target=tuple(), color=None):
+        """Add an arbitrary number of control wires
+
+        Args:
+            layer (Int): the layer to draw the object in
+            wire_ctrl (Union[Int, Iterable[Int]]): set of wires to control on
+            wire_target=tuple() (Union[Int, Iterable[Int]]): target wires. Used to determine min
+                and max wires for the vertical line
+
+        Keyword Args:
+            color=None: mpl compatible color designation
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=2)
+
+            drawer.ctrl(layer=0, wire_ctrl=0, wire_target=1)
+            drawer.ctrl(layer=1, wire_ctrl=(0,1))
+
+        .. figure:: ../../_static/drawer/ctrl.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        wire_ctrl = _to_tuple(wire_ctrl)
+        wire_target = _to_tuple(wire_target)
+
+        wires_all = wire_ctrl + wire_target
+        min_wire = min(wires_all)
+        max_wire = max(wires_all)
+
+        line = plt.Line2D((layer, layer), (min_wire, max_wire), zorder=2, color=color)
+        self.ax.add_line(line)
+
+        for wire in wire_ctrl:
+            circ_ctrl = plt.Circle((layer, wire), radius=self._ctrl_rad, zorder=2, color=color)
+            self.ax.add_patch(circ_ctrl)
+
+    def CNOT(self, layer, wires, color=None):
+        """Draws a CNOT gate.
+
+        Args:
+            layer (Int): layer to draw in
+            wires (Int, Int): tuple of (control, target)
+
+        Keyword Args:
+            color=None: mpl compatible color designation
```

Comment:
```suggestion
            color (None or str): mpl compatible color designation
```
Here and other docstrings.